src/components/CarouselItem.js

```diff
@@ -24,6 +26,21 @@ class CarouselItem extends PureComponent {
     this.props.onTouchStart(event, this.props.index);
   };
 
+  getIsVisible() {
+    if (this.props.carouselIsCentered) {
+      const itemsConsidered = this.props.slidesPerPage % 2 === 0
+        ? this.props.slidesPerPage - 1 : this.props.slidesPerPage;
+
+      const itemsRange = itemsConsidered / 2;
+
+      return this.props.index >= this.props.currentSlideIndex - itemsRange
+        && this.props.index <= this.props.currentSlideIndex + itemsRange;
+    } else {
+      return this.props.index >= this.props.currentSlideIndex
+          && this.props.index < Math.floor(this.props.currentSlideIndex + this.props.slidesPerPage);
```

Comment:
why `Math.floor` as these numbers are always integers?

Reply:
Well, they are not :) I thought it's intentional... I recently migrated 10 or 11 sliders in my project to this carousel, and sometimes I used uneven `slidesPerPage`, which perfectly fit my needs. Try it out on demo page ;)